server/src/test/java/io/crate/lucene/NumericEqQueryTest.java

```diff
@@ -144,4 +154,19 @@ public void test_uses_binary_encoded_range_queries_for_large_numeric() throws Ex
         assertThat(query).isInstanceOf(PointRangeQuery.class);
         assertThat(query.toString()).isEqualTo("y:[-99999999999999999999999999999999999999 TO 274679983711617676]");
     }
+
+    @Test
+    public void test_out_of_bounds_numeric_values_are_filtered_from_numeric_array_literal() {
+        // removing 1.111 because it will never match numeric values with scale = 2; 1.110 is kept since it is equivalent to 1.11
+        assertThat(convert("xarr = [1, 1.1, 1.110, 1.111]::numeric[]").toString())
+            .isEqualTo("+xarr:{100 110 111} +(xarr = [1.0, 1.1, 1.11, 1.111])"); // generic query still contains 1.111
```

Comment:
This assertion fails with actual return, `+xarr:{100 110 111} +(xarr = [1.00, 1.10, 1.11, 1.11])`. The generic function query contains numeric values look to be casted.

My initial attempt to address this was to add a hacky workaround: https://github.com/crate/crate/pull/16829/files#diff-2eaa286cd80d776d19107cfeb1f4d4ed619d12cb7697c5d4109679888526ca59R77-R80.

Could you share your opinions? @seut @matriv

Reply:
Forgot to share, what happens is that as we swap the cast with `SwapCastsInComparisonOperators`, we swap ref.valueType()(numeric(18,2)) not castFunction.valueType()(numeric(null,null)) causing the numeric literal to be casted to numeric(18,2) losing the original scales.

Reply:
Shouldn't we remove the value completely from the generic query?

Reply:
I cannot think of another solution for this one, and couldn't find a case that breaks with this fix in `SwapCastsInComparisonOperators`. Maybe you could also test other variations with `= ANY`, `LIKE` etc.

So, I'd agree to proceed with this fix, but please add a comment there to reveal why it's necessary.

Reply:
Thank you for your input. I played around with `= ANY` but couldn't find anything. `LIKE` seems to not support numeric types(uses text variant) I will follow up with this after.

Reply:
Also GenericFunctionQuery uses NumericType.compare and it correctly handles BigDecimals with different scales.